### PR TITLE
Backport some changes to make 1.18 build with GHC 7.10

### PR DESCRIPTION
**Pull request by ezyang**
This will make it easier to build 7.8 using 7.10

### CODE REVIEW

**Comment by 23Skidoo** [comment]
Fails on Travis.

File: Cabal/Cabal.cabal [hunk 1]
```diff
@@ -43,7 +43,7 @@ extra-source-files:
   tests/PackageTests/BuildDeps/GlobalBuildDepsNotAdditive2/GlobalBuildDepsNotAdditive2.cabal
   tests/PackageTests/BuildDeps/GlobalBuildDepsNotAdditive2/lemon.hs
   tests/PackageTests/BuildDeps/InternalLibrary0/MyLibrary.hs
-  tests/PackageTests/BuildDeps/InternalLibrary0/my.cabal
+  Tests/PackageTests/BuildDeps/InternalLibrary0/my.cabal
```
Comment (tibbe) [inline Cabal/Cabal.cabal]: This looks wrong.

File: cabal-install/cabal-install.cabal [hunk 2]
```diff
@@ -130,7 +130,7 @@ executable cabal
         pretty     >= 1        && < 1.2,
         random     >= 1        && < 1.2,
         stm        >= 2.0      && < 3,
-        time       >= 1.1      && < 1.5,
+        time       >= 1.1      && < 1.6,
```
Comment (tibbe) [inline cabal-install/cabal-install.cabal]: Probably also need to update bootstrap.sh.

**Comment by ezyang** [comment]
OK, fixed, and it looks like it's building now.

**Comment by tibbe** [comment]
@ezyang Cabal itself doesn't install for me using 7.10:

```
~/src/cabal/Cabal (1.18)$ cabal install
Resolving dependencies...
Configuring Cabal-1.18.1.5...
Building Cabal-1.18.1.5...
Failed to install Cabal-1.18.1.5
Build log ( /Users/tibbe/.cabal/logs/Cabal-1.18.1.5.log ):
[ 1 of 70] Compiling Distribution.GetOpt ( Distribution/GetOpt.hs, dist/setup/Distribution/GetOpt.o )
...
[70 of 70] Compiling Main             ( dist/setup/setup.hs, dist/setup/Main.o )
Linking ./dist/setup/setup ...
Configuring Cabal-1.18.1.5...
Building Cabal-1.18.1.5...
Preprocessing library Cabal-1.18.1.5...

on the commandline: Warning:
    -package-name is deprecated: Use -this-package-key instead
ghc: ghc no longer supports single-file style package databases (dist/package.conf.inplace) use 'ghc-pkg init' to create the database with the correct format.
cabal: Error: some packages failed to install:
Cabal-1.18.1.5 failed during the building phase. The exception was:
ExitFailure 1
```

**Pull request merged**